zio-http/src/test/scala/zhttp/http/CookieSpec.scala

```diff
@@ -7,11 +7,24 @@ import zio.test._
 object CookieSpec extends DefaultRunnableSpec {
   def spec = suite("Cookies") {
     suite("response cookies") {
-      testM("encode/decode cookies with ZIO Test Gen") {
-        checkAll(HttpGen.cookies) { cookie =>
-          val cookieString = cookie.encode
-          assert(Cookie.decodeResponseCookie(cookieString))(isSome(equalTo(cookie))) &&
-          assert(Cookie.decodeResponseCookie(cookieString).map(_.encode))(isSome(equalTo(cookieString)))
+      testM("should encode/decode signed/unsigned cookies with same secret") {
+        checkAll(HttpGen.cookies) { case cookie =>
+          cookie.secret match {
+            case Some(secret) => {
+              val cookieString = cookie.encode
+              assert(Cookie.decodeResponseSignedCookie(cookieString, secret))(isSome(equalTo(cookie))) &&
+              assert(
+                Cookie.decodeResponseSignedCookie(cookieString, secret).map(_.encode),
+              )(
+                isSome(equalTo(cookieString)),
+              )
+            }
+            case None         => {
+              val cookieString = cookie.encode
+              assert(Cookie.decodeResponseCookie(cookieString))(isSome(equalTo(cookie))) &&
+              assert(Cookie.decodeResponseCookie(cookieString).map(_.encode))(isSome(equalTo(cookieString)))
+            }
+          }
```

Comment:
I think, we do this https://github.com/dream11/zio-http/pull/751/files#r782728917

We won't need to modify this test much. Only pass the secret as an additional operation.